packages/hub/src/hub/handlers/__test__/handle-wallet-message.test.ts

```diff
@@ -7,15 +14,35 @@ function readSampleMessage(messageName) {
   return require(join(sampleMessagePrefix, messageName + '.json'));
 }
 
-describe('handle-wallet-message', () => {
-  it('ChanelOpen', async () => {
-    const channelOpenAction: ChannelOpen = readSampleMessage('channel-open');
-    const messageReleayRequested = await handleWalletMessage(channelOpenAction);
-    expect(messageReleayRequested).toHaveLength(1);
+async function step<Incoming extends RelayableAction, Outgoing extends RelayableAction>(
+  incoming: string,
+  outgoing: string
+) {
+  const incomingAction: Incoming = readSampleMessage(incoming);
+  const messageReleayRequested1 = await handleWalletMessage(incomingAction);
+  expect(messageReleayRequested1).toHaveLength(1);
+  const expectedAction1: Outgoing = readSampleMessage(outgoing);
+  const receivedAction1 = messageReleayRequested1[0].actionToRelay;
+  expect(receivedAction1).toMatchObject(expectedAction1);
+}
 
-    const expectedAction: ChannelJoined = readSampleMessage('channel-joined');
-    const receivedAction = messageReleayRequested[0].actionToRelay;
-    // note: signatures are not validated
-    expect(receivedAction).toMatchObject(expectedAction);
+describe('handle-wallet-message', () => {
+  // note: signatures are not validated
+  it('open and fund ledger channel sequence', async () => {
+    await step<ChannelOpen, ChannelJoined>('channel-open', 'channel-joined');
```

Comment:
I'm wondering if providing the types adds anything here?

Reply:
It would if there was a way to enforce a type on an object created from requiring a json module. That's something I am planning on looking into.